src/main/asciidoc/transactions.adoc

```diff
@@ -33,82 +41,257 @@ static class Config extends AbstractCouchbaseConfiguration {
-on how to use the `Transactions` class. Since you need access to the current `Collection` as well, we recommend you to also
-autowire the `CouchbaseClientFactory` and access it from there:
+Please see the https://docs.couchbase.com/java-sdk/current/howtos/distributed-acid-transactions-from-the-sdk.html[Reference Documentation]
+
+== Transactions with TransactionTemplate
+The TransactionTemplate class simplifies programmatic transaction demarcation and transaction exception handling.
+The central method is execute, supporting transactional code that implements the TransactionCallback interface. This template handles the transaction lifecycle and possible exceptions such that neither the TransactionCallback implementation nor the calling code needs to explicitly handle transactions.
+Typical usage: Allows for writing low-level data access objects that use resources such as JDBC DataSources but are not transaction-aware themselves. Instead, they can implicitly participate in transactions handled by higher-level application services utilizing this class, making calls to the low-level services via an inner-class callback object.
+Can be used within a service implementation via direct instantiation with a transaction manager reference, or get prepared in an application context and passed to services as bean reference. Note: The transaction manager should always be configured as bean in the application context: in the first case given to the service directly, in the second case given to the prepared template.
+Supports setting the propagation behavior and the isolation level by name, for convenient configuration in context definitions.
+
+.Using the TransactionTemplate
+====
+[source,java]
+----
+// 5.0.0-M5 transaction template bean does not exist, the TransactionTemplate must be explicitly created.
+@Autowired TransactionManager transactionManager;
+TransactionTemplate transactionTemplate = new TransactionTemplate(transactionManager);
+
+// After M6 transaction template bean exists
+@Autowired TransactionTemplate transactionTemplate;
+
+    transactionTemplate.executeWithoutResult(status -> {
+        Person p = personOperations.findById(Person.class).one(id);
+        p.setFirstname(p.getFirstname()+" Junior");
+        personOperations.replaceById(Person.class).one(p);
+    });
+----
+====
+
+== Transactions with CouchbaseTransactionManager
+
+Note: Reactive Couchbase Transactions cannot be supported via the 
+@Transactional annotation. See the section "Transactions with CouchbaseTransactionalOperator"
+for supporting Reactive Couchbase Transactions using the Transactional Operator.
+ 
+@Transactional Describes a transaction attribute on an individual method or on a class.
+ 
+When this annotation is declared at the class level, it applies as a default
+to all methods of the declaring class and its subclasses. Note that it does not
```

Comment:
I feel this documentation tends to lead with these lists of caveats and complexities, which I'd find a bit offputting as a user.  Personally I'd lead with simple examples demo-ing the core concepts, and then if necessary explain and demo the caveats and details after.
Also do we need to repeat complexities and rules of Spring @Transactional in our own docs?  It kinda makes our product looks like it's introducing them, not Spring.  I think it's fine to give a basic overview of @Transactional here, but surely the user should be going to the Spring docs for a true understanding and in particular nuances like this?  It would let us simplify our documentation and focus on what we bring to the @transactional table.

Reply:
I still feel this following the changes :)

Do we have to lead with details of exactly how @Transactional applies to a class hierarchy?  It's straight into the weeds and into yawnsville for the reader... If we have to mention them, could we push them to later?  Or preferable, just figure that we don't need to re-explain how Spring's @Transactional work, and just focus on what we bring to the table?

Reply:
It's more the "lead with" part that was offputting.  The docs are much better on this now after a couple of revisions, but they previously included huge chunks of texts regarding specifics of niche @Transactional attributes, and that kind of thing - all long before we got to just showing how to use it.

I totally accept that sometimes we need to mention some caveats and things for the user to be aware of - like I initially mentioned, "Personally I'd lead with simple examples demo-ing the core concepts, and then if necessary explain and demo the caveats and details after."

I'll comment on the @service thing in that part of the review, but I wasn't looking for an explanation of it as such.